lib/trento/application/integration/checks/checks.ex

```diff
@@ -7,6 +12,52 @@ defmodule Trento.Integration.Checks do
   def request_execution(execution_id, cluster_id, hosts, selected_checks),
     do: adapter().request_execution(execution_id, cluster_id, hosts, selected_checks)
 
+  @spec get_catalog ::
+          {:ok, FlatCatalog.t()} | {:error, any}
+  def get_catalog do
+    case adapter().get_catalog() do
+      {:ok, catalog} ->
+        FlatCatalog.new(%{checks: catalog})
+
+      {:error, :not_ready} ->
+        {:error, "The catalog is still being built. Try again in some moments"}
```

Comment:
it is  preferable to not send strings as errors since they can be hardly matched, here we would just bubble up the error and maybe add a log